domain/relation/state/relation.go

```diff
@@ -425,6 +425,60 @@ AND    e.endpoint_name    = $endpointIdentifier.endpoint_name
 	return corerelation.UUID(uuidAndRole[0].UUID), nil
 }
 
+// GetRelationDetails returns relation details for the given relationID.
+//
+// The following error types can be expected to be returned:
+//   - [relationerrors.RelationNotFound] is returned if the relation UUID
+//     is not found.
+func (st *State) GetRelationDetails(ctx context.Context, relationID int) (relation.RelationDetailsResult, error) {
+	db, err := st.DB()
+	if err != nil {
+		return relation.RelationDetailsResult{}, errors.Capture(err)
+	}
+
+	type getRelation struct {
+		UUID string `db:"uuid"`
+		ID   int    `db:"relation_id"`
+		Life string `db:"value"`
+	}
+	rel := getRelation{
+		ID: relationID,
+	}
+	stmt, err := st.Prepare(`
+SELECT (r.uuid, r.relation_id, l.value) AS (&getRelation.*)
```

Comment:
`SELECT &getRelation.*` works right?

Reply:
It does, but I prefer to be explicit in places like this where they are coming from different tables, especially because life has `id` and relation has `relation_id`.